quesma/queryparser/pancake_transformer.go

```diff
@@ -6,12 +6,43 @@ import (
 	"fmt"
 	"quesma/model"
 	"quesma/model/bucket_aggregations"
+	"reflect"
 	"sort"
 	"strings"
 )
 
 // 2. Translate aggregation tree into pancake model.
 type pancakeTransformer struct {
+	usedNames map[string][]string
+}
+
+// Extremely rarely names may collide (e.g. aggregation named '1__2' and '1', '2').
+// This adds number suffix to make sure they are always unique
+func (a *pancakeTransformer) generateUniqueInternalName(origName string, aggrNames []string) string {
+	if a.usedNames == nil {
+		a.usedNames = make(map[string][]string)
+	}
```

Comment:
```suggestion
func newPancakeTransformer() pancakeTransformer {
    return pancakeTransformer{make(map[string][]string}
}

// Extremely rarely names may collide (e.g. aggregation named '1__2' and '1', '2').
// This adds number suffix to make sure they are always unique
func (a *pancakeTransformer) generateUniqueInternalName(origName string, aggrNames []string) string {
```

Reply:
Nitpick, but I'd change it to this, and use this constructor. This way we don't need to worry about concurrency in this `nil` check.

Reply:
Ok, will do.